src/document-state.js

```diff
@@ -112,6 +116,32 @@ export class DocumentState {
   onVisibilityChanged_() {
     this.visibilityObservable_.fire();
   }
+
+  /**
+   * @param {function()} handler
```

Comment:
Good idea. OOC, on iOS it's encouraged to invoke all callbacks async but I haven't seen that pattern in AMP. Is using setTimeout(f, 0) here bad form?